Add consumer-group rebalance, empty-group, and metadata signals to kafka_consumer (#23915)
DataDog/integrations-core: kafka_consumer/README.md

> - **Brokers**: Configuration and health metrics

Reviewer: Periods introduced in the third bullet with the additional sentence you added, so adding periods to the rest for consistency.

```suggestion
- **Brokers**: Configuration and health metrics.
```

> - **Topics and partitions**: Sizes, offsets, and replication status

Reviewer: ```suggestion
- **Topics and partitions**: Sizes, offsets, and replication status.
```

> - **Consumer groups**: Member details, group state, rebalance detection, membership-change counting, and metadata exposed as tags (`partition_assignor`, `consumer_group_type`, `is_simple_consumer_group`, and `group_instance_id`). Empty/abandoned groups are visible via the `consumer_group_state:EMPTY` tag on `kafka.consumer_group.members`.

Reviewer: ```suggestion
- **Consumer groups**: Member details, group state, rebalance detection, membership-change counting, and metadata exposed as tags (`partition_assignor`, `consumer_group_type`, `is_simple_consumer_group`, and `group_instance_id`). Empty groups are visible through the `consumer_group_state:EMPTY` tag on `kafka.consumer_group.members`.
```

> - **Schema registry**: Schema metadata (requires `schema_registry_url`)

Reviewer: ```suggestion
- **Schema registry**: Schema metadata (requires `schema_registry_url`).
```
